admin/permissions.go

```diff
@@ -167,6 +172,62 @@ func (s *Service) ProjectPermissionsForService(ctx context.Context, projectID, s
 		}, nil
 	}
 
+	// Check if the service has any roles
+	roles, err := s.DB.ResolveProjectRolesForService(ctx, serviceID, projectID)
+	if err != nil {
+		if !errors.Is(err, database.ErrNotFound) {
+			return nil, err
+		}
+	}
+
+	// If roles exist, use them to determine permissions
+	if len(roles) > 0 {
+		composite := &adminv1.ProjectPermissions{}
+		for _, role := range roles {
+			composite = UnionProjectRoles(composite, role)
+		}
+		return composite, nil
+	}
+
+	// If no roles are found, check if the service belongs to the org of the project
+	service, err := s.DB.FindService(ctx, serviceID)
+	if err != nil {
+		return nil, err
+	}
+
+	proj, err := s.DB.FindProject(ctx, projectID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Services get read permissions on the projects of the org they belong to
+	if proj.OrganizationID == service.OrgID {
```

Comment:
I think this needs to be removed now? Otherwise if it has no project roles, it would get access to all projects?

For backwards compatibility for the services with the `admin` org role, I believe the `if orgPerms.ManageProjects` case above covers them.

Reply:
> Otherwise if it has no project roles, it would get access to all projects?

Yes this is intended behaviour, if a service is added as a viewer or any other role than admin to org then it should have read permission on project. Otherwise whats the use of org level roles?

Reply:
made the change